src/main/generic/utils/crypto/CryptoWorkerImpl.js

```diff
@@ -96,11 +96,12 @@ class CryptoWorkerImpl extends IWorker.Stub(CryptoWorker) {
      * @param {Uint8Array} key
      * @param {Uint8Array} salt
      * @param {number} iterations
+     * @param {Hash.Algorithm}
      * @returns {Uint8Array}
      */
-    kdf(key, salt, iterations) {
+    kdf(key, salt, iterations, algorithm = Hash.Algorithm.ARGON2D) {
```

Comment:
Give desired key length here (instead of "hash algorithm", that isn't really a hash algorithm)